src/main/java/com/avispl/symphony/dal/avdevices/camera/panasonic/awue150/CameraPanasonicAWUE150Communicator.java

```diff
@@ -0,0 +1,1779 @@
+	 * @param controls Set of controls that contains AdvancedControllableProperty to be removed
+	 * @param listKeys list key of statistics to be removed
+	 */
+	private void removeUnusedStatsAndControls(Map<String, String> stats, List<AdvancedControllableProperty> controls, Set<String> listKeys) {
+		for (String key : listKeys) {
+			stats.remove(key);
+			controls.removeIf(advancedControllableProperty -> advancedControllableProperty.getName().equals(key));
+		}
+	}
+
+	/**
+	 * Instantiate Text controllable property
+	 *
+	 * @param stats extended statistics
+	 * @param name name of the property
+	 * @param label default button label
+	 * @return AdvancedControllableProperty button instance
+	 */
+	private AdvancedControllableProperty createButton(Map<String, String> stats, String name, String label, String labelPressed) {
+		AdvancedControllableProperty.Button button = new AdvancedControllableProperty.Button();
+		stats.put(name, label);
+		button.setLabel(label);
+		button.setLabelPressed(labelPressed);
+		button.setGracePeriod(0L);
+		return new AdvancedControllableProperty(name, new Date(), button, "");
+	}
+
+	/**
+	 * Create a switch controllable property
+	 *
+	 * @param stats extended statistics
+	 * @param name name of the switch
+	 * @param statusCode initial switch state (0|1)
+	 * @return AdvancedControllableProperty button instance
+	 */
+	private AdvancedControllableProperty createSwitch(Map<String, String> stats, String name, String statusCode, String labelOff, String labelOn) {
+		AdvancedControllableProperty.Switch toggle = new AdvancedControllableProperty.Switch();
+		toggle.setLabelOff(labelOff);
+		toggle.setLabelOn(labelOn);
+		stats.put(name, String.valueOf(statusCode));
+		return new AdvancedControllableProperty(name, new Date(), toggle, statusCode);
+	}
+
+	/***
+	 * Create AdvancedControllableProperty preset instance
+	 *
+	 * @param stats extended statistics
+	 * @param name name of the control
+	 * @param initialValue initial value of the control
+	 * @return AdvancedControllableProperty preset instance
+	 */
+	private AdvancedControllableProperty createDropdown(Map<String, String> stats, String name, List<String> values, String initialValue) {
+		stats.put(name, initialValue);
+		AdvancedControllableProperty.DropDown dropDown = new AdvancedControllableProperty.DropDown();
+		dropDown.setOptions(values.toArray(new String[0]));
+		dropDown.setLabels(values.toArray(new String[0]));
+
+		return new AdvancedControllableProperty(name, new Date(), dropDown, initialValue);
+	}
+
+	/***
+	 * Create AdvancedControllableProperty slider instance
+	 *
+	 * @param stats extended statistics
+	 * @param name name of the control
+	 * @param initialValue initial value of the control
+	 * @return AdvancedControllableProperty slider instance
+	 */
+	private AdvancedControllableProperty createSlider(Map<String, String> stats, String name, String labelStart, String labelEnd, Float rangeStart, Float rangeEnd, Float initialValue) {
```

Comment:
controls creation methods are now a part of core utils, please check com.avispl.symphony.dal.util.ControllablePropertyFactory to see if it fits your needs :)

Reply:
The built-in ControllablePorpertyFactory.createSlider does not Support us to change the label end/ start of the slider, So I won't use it. I only use the createSwitch, createButton, and createDropdown